api/tests/unit/features/workflows/core/test_unit_workflows_models.py

```diff
@@ -773,3 +795,61 @@ def test_change_request_live_from_for_change_request_with_change_set(
 
     # Then
     assert change_request.live_from == now
+
+
+def test_publishing_segments_as_part_of_commit(
+    segment: Segment,
+    change_request: ChangeRequest,
+) -> None:
+    # Given
+    assert segment.version == 2
+    cr_segment = segment.shallow_clone("Test Name", "Test Description", change_request)
+    assert cr_segment.rules.count() == 0
+
+    # Add some rules that the original segment will be cloning from
+    parent_rule = SegmentRule.objects.create(
+        segment=cr_segment, type=SegmentRule.ALL_RULE
+    )
+
+    child_rule1 = SegmentRule.objects.create(
+        rule=parent_rule, type=SegmentRule.ANY_RULE
+    )
+    child_rule2 = SegmentRule.objects.create(
+        rule=parent_rule, type=SegmentRule.NONE_RULE
+    )
+    Condition.objects.create(
+        rule=child_rule1,
+        property="child_rule1",
+        operator=EQUAL,
+        value="condition1",
+        created_with_segment=True,
+    )
+    Condition.objects.create(
+        rule=child_rule2,
+        property="child_rule2",
+        operator=PERCENTAGE_SPLIT,
+        value="0.2",
+        created_with_segment=False,
+    )
+
+    # When
+    change_request._publish_segments()
```

Comment:
Why not just call `change_request.commit()` ?

Reply:
Sure that's no problem